examples/gallery/plot/inset.py

```diff
@@ -0,0 +1,26 @@
+"""
+Inset
+-----
+
+The :meth:`pygmt.Figure.inset` method creates adds an inset figure inside a larger
+figure. The function is called using a ``with`` statement, and it's position,
+border, offset, and margin parameters are set. Within the ``with`` statement,
+pygmt plotting functions can be called that add to the inset figure.
+"""
+import pygmt
+
+fig = pygmt.Figure()
+# Create the primary figure, setting the region to Madagascar, the land color to
+# "brown", the water to "lightblue", the shorelines width to "thin", and adding a frame
+fig.coast(region="MG+r2", land="brown", water="lightblue", shorelines="thin", frame="a")
+# Create an inset, setting the position to top left, the width to 3.5 centimeters, and
+# the x-offset to 0.2 centimeters. The margin is set to 0, and the border is "green".
+with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen"):
+    # Create a figure in the inset using coast. This example uses the azimuthal
+    # orthogonal projection centered at 47E, 20S. The 4 centimeter width of this figure
+    # extends beyond the inset width, and isn't shown. The land is set to "gray" and
+    # Madagascar is highlighted in "red".
+    fig.coast(
+        region="g", projection="G47/-20/?c", land="gray", water="white", dcw="MG+gred"
+    )
```

Comment:
Something's wrong with the inset coloring of Madagascar here, can you double-check?

![image](https://user-images.githubusercontent.com/23487320/106434410-61b59080-64d6-11eb-9981-db46eb908e6b.png)

Reply:
I think this may have to do with using "M?c" for the projection; I'll test the old formatting and see if that fixes it.

Reply:
@weiji14 Think I fixed it in with 7ecffd1

Reply:
> Something's wrong with the inset coloring of Madagascar here, can you double-check?

I don't get it. What's the problem? Why changing `M?c` to `M3.5c` can fix it?

Reply:
The issue is that the dcw argument doesn't paint in the correct area on the inset. I'm not sure how specifying the width fixed the problem, but it appears to have done so.

Reply:
Oh yes, you're right. Please forgive my bad knowledge about the location of Madagascar.

It looks like an upstream GMT bug to me. Do you want to try to reproduce the issue using GMT CLI? If you can reproduce it using GMT CLI, then we should report it to upstream

Reply:
If you want to try, you can follow this comment (https://github.com/GenericMappingTools/pygmt/issues/573#issuecomment-701063997) to create the equivalent GMT script. Otherwise, I'll do it later.

Reply:
I'll give it a go; about time I relearned how to use actual GMT!

Reply:
I've reported the bug to upstream https://github.com/GenericMappingTools/gmt/issues/4757